test/integration/controller/core/clusterqueue_controller_test.go

```diff
@@ -603,4 +610,146 @@ var _ = ginkgo.Describe("ClusterQueue controller", func() {
 			}, util.Timeout, util.Interval).Should(testing.BeNotFoundError())
 		})
 	})
+
+	ginkgo.When("Reconciling clusterQueue pending workload status", func() {
+		var (
+			clusterQueue   *kueue.ClusterQueue
+			localQueue     *kueue.LocalQueue
+			onDemandFlavor *kueue.ResourceFlavor
+		)
+
+		ginkgo.BeforeEach(func() {
+			onDemandFlavor = testing.MakeResourceFlavor(flavorOnDemand).Obj()
+			gomega.Expect(k8sClient.Create(ctx, onDemandFlavor)).To(gomega.Succeed())
+			clusterQueue = testing.MakeClusterQueue("cluster-queue").
+				ResourceGroup(
+					*testing.MakeFlavorQuotas(flavorOnDemand).
+						Resource(corev1.ResourceCPU, "5", "5").Obj(),
+				).
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, clusterQueue)).To(gomega.Succeed())
+			localQueue = testing.MakeLocalQueue("queue", ns.Name).ClusterQueue(clusterQueue.Name).Obj()
+			gomega.Expect(k8sClient.Create(ctx, localQueue)).To(gomega.Succeed())
+		})
+
+		ginkgo.AfterEach(func() {
+			util.ExpectClusterQueueToBeDeleted(ctx, k8sClient, clusterQueue, true)
+			util.ExpectResourceFlavorToBeDeleted(ctx, k8sClient, onDemandFlavor, true)
+		})
+
+		ginkgo.It("Should update of the pending workloads when a new workload is scheduled", func() {
+			workloads := []*kueue.Workload{
+				testing.MakeWorkload("one", ns.Name).Queue(localQueue.Name).
+					Request(corev1.ResourceCPU, "2").Request(resourceGPU, "2").Obj(),
+				testing.MakeWorkload("two", ns.Name).Queue(localQueue.Name).
+					Request(corev1.ResourceCPU, "3").Request(resourceGPU, "3").Obj(),
+			}
+
+			ginkgo.By("Creating workloads")
+			for _, w := range workloads {
+				gomega.Expect(k8sClient.Create(ctx, w)).To(gomega.Succeed())
+			}
+			gomega.Eventually(func() *kueue.ClusterQueuePendingWorkloadsStatus {
+				var updatedCq kueue.ClusterQueue
+				gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(clusterQueue), &updatedCq)).To(gomega.Succeed())
+				return updatedCq.Status.PendingWorkloadsStatus
+			}, util.Timeout, util.Interval).Should(gomega.BeComparableTo(&kueue.ClusterQueuePendingWorkloadsStatus{
+				Head: []kueue.ClusterQueuePendingWorkload{
+					{
+						Name:      "one",
+						Namespace: ns.Name,
+					},
+					{
+						Name:      "two",
+						Namespace: ns.Name,
+					},
+				},
+			},
+				ignoreLastChangeTime,
+				cmpopts.SortSlices(func(i, j kueue.ClusterQueuePendingWorkload) bool {
+					return i.Name < j.Name
+				})))
+
+			ginkgo.By("Creating a new workload")
+			newWorkloads := []*kueue.Workload{
+				testing.MakeWorkload("three", ns.Name).Queue(localQueue.Name).
+					Request(corev1.ResourceCPU, "2").Request(resourceGPU, "2").Obj(),
+				testing.MakeWorkload("four", ns.Name).Queue(localQueue.Name).
+					Request(corev1.ResourceCPU, "3").Request(resourceGPU, "3").Obj(),
+			}
+			for _, w := range newWorkloads {
+				gomega.Expect(k8sClient.Create(ctx, w)).To(gomega.Succeed())
+				time.Sleep(time.Second)
```

Comment:
Why is this sleep needed? Is it because otherwise the workload "four" may be created in the same second as workload "one" and thus, the cut off would not be deterministic? If this is the case I think this is too hidden. I would prefer to use workload priorities to make sure the workload cut off is deterministic.

Reply:
yes, exactly, what do you mean under workload priorities?

Reply:
it was removed when workload priority was add https://github.com/kubernetes-sigs/kueue/pull/1069/commits/3f01af197b7d150455e4421a70ae1c12cf59c0e6